mono/metadata/object.c

```diff
@@ -54,6 +54,13 @@
 #include "cominterop.h"
 #include <mono/utils/w32api.h>
 #include <mono/utils/unlocked.h>
+#include "external-only.h"
+#include "monitor.h"
+
+// Exports do not work portably/reliably if compiled separately.
+// They must be in .o/.obj files otherwise referenced.
+// .def files work better in this regard.
+#include "external-only.c"
```

Comment:
Discussed with Jay.  `#include "external-only.c"` can go away.  We don't quite have the problem like in https://github.com/mono/mono/pull/11188/files#r226004001 because we don't build intermediate `.lib` on Windows.  We should revisit using `.def` files another day.